pkg/controller/namespace/namespace_controller_utils.go

```diff
@@ -154,7 +156,12 @@ func deleteCollection(
 	}
 
 	apiResource := unversioned.APIResource{Name: gvr.Resource, Namespaced: true}
-	err := dynamicClient.Resource(&apiResource, namespace).DeleteCollection(nil, &v1.ListOptions{})
+
+	// namespace controller does not want the garbage collector to insert the orphan finalizer since it calls
+	// resource deletions generically.  it will ensure all resources in the namespace are purged prior to releasing
+	// namespace itself.
+	orphanDependents := false
+	err := dynamicClient.Resource(&apiResource, namespace).DeleteCollection(&v1.DeleteOptions{OrphanDependents: &orphanDependents}, &v1.ListOptions{})
```

Comment:
DeleteCollection delegates to the ordinary Delete call:
https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/generic/registry/store.go#L788

So it should work as expected.